src/components/Card.js

```diff
@@ -38,19 +55,27 @@ export default class Card {
       this._handleLikeButton();
     });
     this._cardDeleteButton.addEventListener("click", () => {
-      this._handleDeleteButton();
+      this._deleteCardConfirm(this._id, this._cardElement);
     });
     this.cardImageElement.addEventListener("click", () => {
       this._handleImageClick({ link: this._link, name: this._name });
     });
   }
 
-  _handleDeleteButton() {
-    this._cardElement.remove();
-    this._cardElement = null;
-  }
+  // handleDeletion() {
+  //   this._cardElement.remove();
+  //   this._cardElement = null; //use/call these after the delete has been confirmed?
+  // }
 
   _handleLikeButton() {
-    this._cardLikeButton.classList.toggle("card__like-button-active");
+    if (!this._isLiked) {
+      this._cardLikeButton.classList.add("card__like-button-active");
```

Comment:
You need to **toggle likes** only in block `then` after a successful response from the server because if there is an error the user will not know about that and will see the toggled heart color.